docs/move_clusters.md

```diff
@@ -0,0 +1,59 @@
+# Moving Hive-managed Clusters
+
+## Summary
+
+There is a need to be able to move Hive-managed clusters from one Hive instance to another.
+This may be useful if the user want to replace a Hive instance with a new one or if the user
+wants to implement sharding between Hive instances.
+
+Hive will manage all of the moving for the user. The user will create a CR indicating with
+details about the destination cluster and label selectors to match against the cluster to
+move.
+
+## ClusterRelocator CRD
+
+A ClusterRelocator is cluster-scoped.
+
+This is an example of a ClusterRelocator that may be used to move all of the ClusterDeployments
+with the label `hive-instance: hive2` to the remote cluster accessed via the kubeconfig secret
+at "hive2-remote-kubecofig-secret" in the "hive" namespace.
```

Comment:
@jmelis raised a good point here, are we ok for networking? Will this require vpc peering to get from priv cluster to priv cluster?

CC @mwoodson @twiest

Reply:
Are you suggesting that the Hive instances would be run in 2 separate private clusters? The relocation does not interact with the cluster created by Hive.

Reply:
Or are you suggesting that VPC peering would be set up so that the first Hive instance can communicate with the private cluster and would need to be changed so that the second Hive instance can communicate with the private cluster? I do assume that there will be other parts outside of Hive that will need to do work to relocate.

Reply:
Each Hive instance in a separate private cluster, connectivity between the two may not be a given to do the push from A to B may not be a given and require some additional thought.

Reply:
OK. Well that is a major restriction that would make this entire proposal unworkable. So I will stop moving forward on any implementation of this until we can get an answer on whether it is feasible.

Reply:
It should be fine, it sounds like it's just VPC peering but just need to hear back from those involved if that is feasible.

Reply:
cc @pbergene and @rogbas too
> @jmelis raised a good point here, are we ok for networking? Will this require vpc peering to get from priv cluster to priv cluster?

Reply:
And to clarify, no need to stop on impl, we just need to coordinate. This is the best option as it lets us rebalance as well as migrate, we don't want anything else IMO.

Reply:
I agree, I don't expect any major issues with this VPC peering, but it needs some thinking as @dgoodwin has mentioned